gamestonk_terminal/cryptocurrency/discovery/pycoingecko_model.py

```diff
@@ -88,458 +74,214 @@
-        change_1h, change_24h, _ = changes_parser(changes)
-        results.append([name, symbol, price, change_1h, change_24h, last_added, url])
-    df = replace_qm(pd.DataFrame(results, columns=columns))
-    df.index = df.index + 1
-    df.reset_index(inplace=True)
-    df.rename(columns={"index": "Rank"}, inplace=True)
-    df["Price"] = df["Price"].apply(lambda x: float(x.strip("$").replace(",", "")))
-    return df
-
-
-# This function does not use coingecko api because there is not an endpoint for this
-def get_yield_farms() -> pd.DataFrame:
-    """Scrapes yield farms data from "https://www.coingecko.com/en/yield-farming" [Source: CoinGecko]
-
+    top: int
+        Number of top coins to grab from CoinGecko
     Returns
     -------
     pandas.DataFrame
-        Top Yield Farms
-        Columns: Rank, Name, Pool, Audits, Collateral, Value Locked, Return Year, Return Hour
+        N coins
     """
-
-    columns = [
-        "Rank",
-        "Name",
-        "Pool",
-        "Audits",
-        "Collateral",
-        "Value_Locked",
-        "Return_Year",
-    ]
-    url = "https://www.coingecko.com/en/yield-farming"
-    try:
-        scraped_data = scrape_gecko_data(url)
-    except RetryError as e:
-        console.print(e)
-        return pd.DataFrame()
-    rows = scraped_data.find("tbody").find_all("tr")
-    results = []
-    for row in rows:
-        row_cleaned = clean_row(row)[:-2]
-        if " New" in row_cleaned:  # find better way to fix it in future
-            row_cleaned.remove(" New")
-
-        if len(row_cleaned) == 7:
-            row_cleaned.insert(2, None)
-        (
-            rank,
-            name,
-            pool,
-            *others,
-            _,
-            value_locked,
-            apy1,
-            _,  # hourly removed for most cases it's 0.00 so it doesn't bring any value for user
-        ) = row_cleaned
-        auditors, collateral = collateral_auditors_parse(others)
-        auditors = ", ".join(aud.strip() for aud in auditors)
-        collateral = ", ".join(coll.strip() for coll in collateral)
-        results.append(
-            [
-                rank,
-                name,
-                pool,
-                auditors,
-                collateral,
-                value_locked,
-                apy1,
-            ]
-        )
-    df = pd.DataFrame(results, columns=columns).replace({"": None})
-    for col in ["Return_Year"]:
-        df[col] = df[col].apply(
-            lambda x: x.replace(" Yearly", "") if isinstance(x, str) else x
-        )
-    df["Rank"] = df["Rank"].astype(int)
-    df = wrap_text_in_df(df, w=30)
+    i = 1
+    remaining_top = top
+    client = CoinGeckoAPI()
+    if category:
+        categories_dict = read_file_data("coingecko_categories.json")
+        if category not in categories_dict:
+            raise ValueError(
+                f"Category does not exist\nPlease chose one from list: {categories_dict.keys()}"
+            )
+    if top > 250:
```

Comment:
There's a lot of repeated code here, there must be a way to have it more clean and more readable. Let me know if you need help in that!

Reply:
yea i think this was the best spaghetti function i ever did in python 🍝